### PR TITLE
httpcaddyfile: Fix unexpectedly removed policy

### PR DESCRIPTION
**Pull request by skeetmtp**
When user set on_demand tls option in a catch-all (:443) policy,
we expect other policies to not have the on_demand enabled
See ex in tls_automation_policies_5.txt

Btw, we can remove policies if they are **all** empty.

### CODE REVIEW

**Comment by skeetmtp** [comment]
For your information, the following txt show the current caddy behaviour : 
```
a.example.com {
}

b.example.com {
}

:443 {
	tls {
		on_demand
	}
}
----------
{
	"apps": {
		"http": {
			"servers": {
				"srv0": {
					"listen": [
						":443"
					],
					"routes": [
						{
							"match": [
								{
									"host": [
										"a.example.com"
									]
								}
							],
							"terminal": true
						},
						{
							"match": [
								{
									"host": [
										"b.example.com"
									]
								}
							],
							"terminal": true
						}
					]
				}
			}
		},
		"tls": {
			"automation": {
				"policies": [
					{
						"on_demand": true
					}
				]
			}
		}
	}
}
```

**Comment by skeetmtp** [comment]
In the mean time, a workaround is to put a global tls option, so the "remove empty AP" loop do not do anything.
```

{
    email me@company.com
}
...
```

**Comment by francislavoie** [comment]
Thanks -- please don't forget to sign the CLA https://github.com/caddyserver/caddy/pull/4128#issuecomment-824926843

File: caddyconfig/httpcaddyfile/tlsapp.go [hunk 1]
```diff
@@ -480,15 +480,19 @@ func consolidateAutomationPolicies(aps []*caddytls.AutomationPolicy) []*caddytls
 		return len(aps[i].Subjects) > len(aps[j].Subjects)
 	})
 
-	// remove any empty policies (except subjects, of course)
+	emptyAPCount := 0
+	// compute the number of empty policies (except subjects, of course)
```
Comment (mholt) [inline caddyconfig/httpcaddyfile/tlsapp.go:484]:
```suggestion
	// compute the number of empty policies (disregarding subjects) - see #4128
```

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks for the fix! Looks simple enough. Somewhere in the comments, we should refer to this PR. I made one suggestion for that.

**Review by mholt — APPROVED** [review]

**Pull request merged**